### PR TITLE
Add time structure.

### PR DESCRIPTION
**Pull request by Shylock-Hg**

### CODE REVIEW

**Comment by panda-sheep** [comment]
About `time`, I am more curious.
Is there a requirement  to use `time`?
I think `time `in the database is a feature  that will be obsolete

**Review by dutor — COMMENTED** [review]
Well done.

As we discussed, `Time` and `DateTime` should not consist of timezone.

**Comment by Shylock-Hg** [comment]
> About `time`, I am more curious.
> Is there a requirement to use `time`?
> I think `time `in the database is a feature that will be obsolete

Many database has it, like mysql, neo4j.

**Comment by Shylock-Hg** [comment]
> Well done.
> 
> As we discussed, `Time` and `DateTime` should not consist of timezone.

Fixed.

File: src/common/datatypes/Value.h [hunk 1]
```diff
@@ -350,9 +362,12 @@ struct Value {
     // Date value
     void setD(const Date& v);
     void setD(Date&& v);
+    // Date value
+    void setTm(const Time& v);
+    void setTm(Time&& v);
     // DateTime value
-    void setT(const DateTime& v);
-    void setT(DateTime&& v);
+    void setDt(const DateTime& v);
```
Comment (laura-ding) [inline src/common/datatypes/Value.h:369]: setDT

File: src/common/datatypes/Value.h [hunk 2]
```diff
@@ -350,9 +362,12 @@ struct Value {
     // Date value
     void setD(const Date& v);
     void setD(Date&& v);
+    // Date value
+    void setTm(const Time& v);
```
Comment (laura-ding) [inline src/common/datatypes/Value.h:366]: setT

File: src/common/datatypes/Value.h [hunk 3]
```diff
@@ -312,7 +323,8 @@ struct Value {
         double                      fVal;
         std::string                 sVal;
         Date                        dVal;
-        DateTime                    tVal;
+        Time                        tmVal;
```
Comment (laura-ding) [inline src/common/datatypes/Value.h:326]: tVal

File: src/common/datatypes/Value.h (hunk 1 above)
Comment (Shylock-Hg) [reply src/common/datatypes/Value.h]: Fixed.

File: src/common/datatypes/Value.h (hunk 2 above)
Comment (Shylock-Hg) [reply src/common/datatypes/Value.h]: Fixed.

File: src/common/datatypes/Value.h (hunk 3 above)
Comment (Shylock-Hg) [reply src/common/datatypes/Value.h]: Fixed.

File: src/common/interface/common.thrift [hunk 4]
```diff
@@ -46,16 +46,25 @@ typedef i32 (cpp.type = "nebula::IndexID") IndexID
...
-
-// !! Struct DateTime has a shadow data type defined in the ThriftTypes.h
+// !! Struct Time has a shadow data type defined in the Date.h
+// So any change here needs to be reflected to the shadow type there
+struct Time {
+    1: byte hour;         // Hour: 0 - 23
+    2: byte minute;       // Minute: 0 - 59
+    3: byte sec;          // Second: 0 - 59
+    4: i32 microsec;    // Micro-second: 0 - 999,999
+    5: i32 timezone;    // Time difference in seconds
```
Comment (laura-ding) [inline src/common/interface/common.thrift:64]: delete the `timezone`
the `DateTime` as the same

Comment (Shylock-Hg) [reply src/common/interface/common.thrift]: Good catch. Fixed.

**Review by laura-ding — APPROVED** [review]

**Review by dutor — APPROVED** [review]

**Pull request merged**